Fixes #32635 - fix already initialized constant

lzap opened the pull request:
I am seing the following warning during startup:

```
app/services/facets.rb:2: warning: already initialized constant Facets::SUPPORTED_CORE_OBJECTS
app/services/facets.rb:2: warning: previous definition of SUPPORTED_CORE_OBJECTS was here
```

To reproduce, you might need to enable foreman_webhooks which performs eager loading in the plugin initializer to get list of all classes.

This change fixes the issue, however I do not know what consequences of this change might be. We do use `require_dependency` pretty much everywhere in the codebase for our files, except for facets. Was this intentional? Perhaps not.

lzap committed:
Fixes #32635 - fix already initialized constant

lzap commented:
@ShimShtein

lzap commented on config/application.rb:
```diff
@@ -100,15 +100,13 @@ class Application < Rails::Application
+    config.autoload_paths += %W(#{config.root}/app/models/power_manager)
+    config.autoload_paths += %W(#{config.root}/app/models/concerns)
+    config.autoload_paths += %W(#{config.root}/app/services)
+    config.autoload_paths += %W(#{config.root}/app/mailers)
```
No change here I am just rewriting it so all lines in this block use the same syntax.

From my testing, it looks like autoloading, at least in Rails version which is in use, is recursive. I created `lib/test/test.rb` and it autoloaded the class from there.

ezr-ondrej replied:
it definitelly looks much better, I thing that this would look even better, but that's only my preference.
```
config.autoload_paths << config.root.join('app', 'mailers')
```

ezr-ondrej commented on config/application.rb:
```diff
@@ -100,15 +100,13 @@ class Application < Rails::Application
-
+    # Autoloading
+    config.autoload_paths += %W(#{config.root}/lib)
+    config.autoload_paths += %W(#{config.root}/app/controllers/concerns)
```
This should be automatically added by rails.

ezr-ondrej commented on config/application.rb:
```diff
@@ -100,15 +100,13 @@ class Application < Rails::Application
+    config.autoload_paths += %W(#{config.root}/lib)
+    config.autoload_paths += %W(#{config.root}/app/controllers/concerns)
+    config.autoload_paths += %W(#{config.root}/app/models/power_manager)
+    config.autoload_paths += %W(#{config.root}/app/models/concerns)
```
ditto ^^

ezr-ondrej commented on config/application.rb:
```diff
@@ -100,15 +100,13 @@ class Application < Rails::Application
+    config.autoload_paths += %W(#{config.root}/app/controllers/concerns)
+    config.autoload_paths += %W(#{config.root}/app/models/power_manager)
+    config.autoload_paths += %W(#{config.root}/app/models/concerns)
+    config.autoload_paths += %W(#{config.root}/app/services)
```
I believe this is also included by Rails by `app/*` (all subdirectories to app should be included)

ezr-ondrej reviewed (commented):
Thank you @lzap for taking care of this, I'd love to clean it up a bit, as a lot of our autoload paths are automatically added by Rails.
Maybe it was not the case 8 years ago and that's why we have them around

ShimShtein commented:
@lzap, do you also want to take care of:
```
/home/vagrant/foreman/lib/foreman.rb:8: warning: already initialized constant Foreman::UUID_REGEXP
/home/vagrant/foreman/app/services/foreman/env_settings_loader.rb:3: warning: already initialized constant Foreman::EnvSettingsLoader::LOGGERS_FROM_ENV_REGEX
```

lzap commented:
Done all remarks, also fixed the other constants. I do not understand why I saw them, during debugging they were always required just once. Therefore I converted them into methods:

https://medium.com/invoca-engineering-blog/ruby-constants-with-rails-and-how-to-avoid-the-already-initialized-constant-x-warning-753d28f17691

ezr-ondrej reviewed (approved):
Looks good, works as expected, thanks @lzap !
